### PR TITLE
Add Semantic API and Language server updates to release note of 2201.3.0

### PR DESCRIPTION
**Pull request by dulajdilshan**
$subject

- Semantic API: [https://github.com/ballerina-platform/ballerina-lang/issues?q=is%3Aissue+is%3Aclosed+label%3AArea%2FSemanticAPI+milestone%3A2201.3.0](https://github.com/ballerina-platform/ballerina-lang/issues?q=is%3Aissue+is%3Aclosed+label%3AArea%2FSemanticAPI+milestone%3A2201.3.0) 

- Lang Server: [https://github.com/ballerina-platform/ballerina-lang/issues?q=is%3Aissue+is%3Aclosed+milestone%3A2201.3.0+label%3ATeam%2FLanguageServer](https://github.com/ballerina-platform/ballerina-lang/issues?q=is%3Aissue+is%3Aclosed+milestone%3A2201.3.0+label%3ATeam%2FLanguageServer)

### CODE REVIEW

File: release-notes/swan-lake-release-notes/swan-lake-2201.3.0-release-note.md [hunk 1]
```diff
@@ -220,6 +237,12 @@ To view bug fixes, see the [GitHub milestone for 2201.x.0 (Swan Lake)](https://g
 
 ### New features
 
+#### Language Server
+
+- Introduced new `Extract to local variable`code action
```
Comment (TharushiJay) [inline release-notes/swan-lake-release-notes/swan-lake-2201.3.0-release-note.md:242]: What about `Extract to function` code action? https://github.com/ballerina-platform/ballerina-lang/issues/36570

File: release-notes/swan-lake-release-notes/swan-lake-2201.3.0-release-note.md [hunk 2]
```diff
@@ -256,6 +284,7 @@ To view bug fixes, see the GitHub milestone for 2201.3.0 (Swan Lake) of the repo
 - [Update Tool](https://github.com/ballerina-platform/ballerina-update-tool/issues?q=is%3Aissue+milestone%3A%22Ballerina+2201.3.0%22+is%3Aclosed+label%3AType%2FBug)
```
Comment (mohanvive) [inline release-notes/swan-lake-release-notes/swan-lake-2201.3.0-release-note.md:284]: Language server link should be corrected as `https://github.com/ballerina-platform/ballerina-lang/issues?q=is%3Aissue+milestone%3A2201.3.0+label%3AType%2FBug+label%3ATeam%2FLanguageServer`

File: release-notes/swan-lake-release-notes/swan-lake-2201.3.0-release-note.md [hunk 3]
```diff
@@ -220,6 +237,13 @@ To view bug fixes, see the [GitHub milestone for 2201.x.0 (Swan Lake)](https://g
 
 ### New features
 
+#### Language Server
+
+- Introduced new `Extract to local variable` code action
+- Introduced new `Extract to function` code action
+- Introduced `loadProject()` API to the workspace manager
```
Comment (mohanvive) [inline release-notes/swan-lake-release-notes/swan-lake-2201.3.0-release-note.md:244]: Introduced `loadProject()` API to the workspace manager to open up a project programmatically.

**Review by mohanvive — DISMISSED** [review]

File: release-notes/swan-lake-release-notes/swan-lake-2201.3.0-release-note.md [hunk 4]
```diff
@@ -42,6 +42,23 @@ If you have not installed Ballerina, then download the [installers](/downloads/#
 
 To view bug fixes, see the [GitHub milestone for 2201.x.0 (Swan Lake)](https://github.com/ballerina-platform/ballerina-lang/issues?q=is%3Aissue+is%3Aclosed+label%3AType%2FBug+label%3ATeam%2FCompilerFE+milestone%3A%22Ballerina+2201.3.0%22).
 
+## Compiler API updates
+
+### New features
+
+#### Semantic API
+- Added a new API: `annotAttachments()` to get the annotation attachments and their constant values from the annotatable symbols
```
Comment (praneesha) [inline release-notes/swan-lake-release-notes/swan-lake-2201.3.0-release-note.md:50]:
```suggestion
- Added a new `annotAttachments()` API to get the annotation attachments and their constant values from the annotatable symbols
```

File: release-notes/swan-lake-release-notes/swan-lake-2201.3.0-release-note.md [hunk 5]
```diff
@@ -42,6 +42,23 @@ If you have not installed Ballerina, then download the [installers](/downloads/#
...
+- Introduced a new symbol: `ClientDeclSymbol` to represent semantic information of client-declaration statement
```
Comment (praneesha) [inline release-notes/swan-lake-release-notes/swan-lake-2201.3.0-release-note.md:51]:
```suggestion
- Introduced a new `ClientDeclSymbol` symbol to represent the semantic information of the client-declaration statement
```

File: release-notes/swan-lake-release-notes/swan-lake-2201.3.0-release-note.md [hunk 6]
```diff
@@ -42,6 +42,23 @@ If you have not installed Ballerina, then download the [installers](/downloads/#
...
+
+### Improvements
+
+#### Semantic API
+- Improved `constValue()` method to retrieve the constant value as an object from the constant-symbol
```
Comment (praneesha) [inline release-notes/swan-lake-release-notes/swan-lake-2201.3.0-release-note.md:56]:
```suggestion
- Improved the `constValue()` method to retrieve the constant value as an object from the constant symbol
```

File: release-notes/swan-lake-release-notes/swan-lake-2201.3.0-release-note.md [hunk 7]
```diff
@@ -42,6 +42,23 @@ If you have not installed Ballerina, then download the [installers](/downloads/#
...
+
+### Bug fixes
+
+To view bug fixes of Semantic API, see the [GitHub milestone for 2201.3.0 (Swan Lake): Semantic API](https://github.com/ballerina-platform/ballerina-lang/issues?q=is%3Aissue+milestone%3A2201.3.0+label%3ATeam%2FCompilerFETools+label%3AType%2FBug+).
```
Comment (praneesha) [inline release-notes/swan-lake-release-notes/swan-lake-2201.3.0-release-note.md:60]:
```suggestion
To view bug fixes, see the [GitHub milestone for 2201.3.0 (Swan Lake)](https://github.com/ballerina-platform/ballerina-lang/issues?q=is%3Aissue+milestone%3A2201.3.0+label%3ATeam%2FCompilerFETools+label%3AType%2FBug+).
```
Updated to be consistent with the other sections of the release note template.

File: release-notes/swan-lake-release-notes/swan-lake-2201.3.0-release-note.md [hunk 8]
```diff
@@ -220,6 +237,13 @@ To view bug fixes, see the [GitHub milestone for 2201.x.0 (Swan Lake)](https://g
 
 ### New features
 
+#### Language Server
+
+- Introduced new `Extract to local variable` code action
```
Comment (praneesha) [inline release-notes/swan-lake-release-notes/swan-lake-2201.3.0-release-note.md:242]:
```suggestion
- Introduced a new `Extract to local variable` code action
```

File: release-notes/swan-lake-release-notes/swan-lake-2201.3.0-release-note.md [hunk 9]
```diff
@@ -220,6 +237,13 @@ To view bug fixes, see the [GitHub milestone for 2201.x.0 (Swan Lake)](https://g
...
+- Introduced new `Extract to function` code action
```
Comment (praneesha) [inline release-notes/swan-lake-release-notes/swan-lake-2201.3.0-release-note.md:243]:
```suggestion
- Introduced a new `Extract to function` code action
```

File: release-notes/swan-lake-release-notes/swan-lake-2201.3.0-release-note.md [hunk 10]
```diff
@@ -220,6 +237,13 @@ To view bug fixes, see the [GitHub milestone for 2201.x.0 (Swan Lake)](https://g
...
+- Introduced `loadProject()` API to the workspace manager to open up a project programmatically
```
Comment (praneesha) [inline release-notes/swan-lake-release-notes/swan-lake-2201.3.0-release-note.md:244]:
```suggestion
- Introduced the `loadProject()` API to the workspace manager to open up a project programmatically
```

File: release-notes/swan-lake-release-notes/swan-lake-2201.3.0-release-note.md [hunk 11]
```diff
@@ -220,6 +237,13 @@ To view bug fixes, see the [GitHub milestone for 2201.x.0 (Swan Lake)](https://g
...
+- Introduced a new code action to generate module for client declarations (`Generate module for client declaration`)
```
Comment (praneesha) [inline release-notes/swan-lake-release-notes/swan-lake-2201.3.0-release-note.md:245]:
```suggestion
- Introduced a new `Generate module for client declaration` code action to generate a module for client declarations
```

File: release-notes/swan-lake-release-notes/swan-lake-2201.3.0-release-note.md [hunk 12]
```diff
@@ -240,6 +264,10 @@ Introduced the `--native` flag, which generates a GraalVM native executable when
 
 ### Improvements
 
+#### Language Server
+- Improved completions sorting within the expression of the `if` condition
+- Improved `Change variable type` code action
```
Comment (praneesha) [inline release-notes/swan-lake-release-notes/swan-lake-2201.3.0-release-note.md:269]:
```suggestion
- Improved the `Change variable type` code action
```

**Review by malinthar — APPROVED** [review]

**Pull request merged**